### PR TITLE
Make the `Value` indexable

### PR DESCRIPTION
**Pull request by temeddix**
This PR does just what the title says

With this PR, operations like below are made possible:
```rust
let value:Value = deserialize(bytes).unwrap();
let inner:Value = value["someItem"]["breads"][3]; 
```

### CODE REVIEW

**Comment by nyuichi** [comment]
Looks pretty good. I like the direction this pr is going forward.
I want some test cases, docs, and more impls like what serde_json has, so could you please be waited until I finish fixing up?

**Comment by nyuichi** [comment]
@temeddix 
May I have a permission to push to your repository?

**Comment by temeddix** [comment]
@nyuichi Of course :D I invited you as a collaborator, I will boost the permission after the invitation is accepted.

**Comment by nyuichi** [comment]
Mmm, seems like I've got into trouble regarding this 🤔  https://docs.github.com/en/codespaces/codespaces-reference/security-in-github-codespaces#working-with-other-peoples-contributions-and-repositories

**Comment by temeddix** [comment]
Hmm, as far as I know repository maintainers can suggest PR changes by writing the code themselves

**Comment by nyuichi** [comment]
I managed to push it :)

@umadein Can you review it? If you're quite busy I'll leave it maybe for 3 days or so and merge it without approval then.

**Comment by nyuichi** [comment]
I will release 0.8 once this gets merged.

**Review by umadein — APPROVED** [review]
LGTM!

**Pull request merged**
